Mac-App/Mac-App/Transformers/DetailsViewTransformer.swift

```diff
@@ -25,30 +40,38 @@ final class DetailsViewTransformer: StoreSubscriber {
         }
     }
 
-    private func mapAppStateToViewData(_ appState: AppState) -> DetailsViewTransformer.Status {
-        guard appState.dependencyGraphState.tree.count > 0 else {
+    private func mapAppStateToViewData(_ data: ProjectFactsViewData.Data) -> DetailsViewTransformer.Status {
+        guard data.totalDependenciesFound > 0 else {
             return .failure("No paths found!")
         }
 
+        return .success(data)
+    }
+}
+
+extension DetailsViewTransformer {
+    enum Status {
+        case success(_ data: ProjectFactsViewData.Data)
+        case failure(_ failure: String)
+    }
+}
+
+// MARK: - Mapper from AppState to subscriber state (view data for UI)
+
+extension ProjectFactsViewData.Data {
+    init(appState: AppState) {
         let heaviestDependency = appState.dependencyGraphState.tree.sorted(by: { first, second -> Bool in
             first.dependencies.count > second.dependencies.count
-            }).first!.owner
+            }).first?.owner
         let totalDeps = appState.dependencyGraphState.tree.count
         let paths = appState.dependencyPathsState.paths
 
-        let data = ProjectFactsViewData.Data(
-            heaviestDependency: heaviestDependency,
+        self.init(
+            heaviestDependency: heaviestDependency ?? "",
             totalDependenciesFound: totalDeps,
             paths: paths
         )
-
-        return .success(data)
     }
 }
 
-extension DetailsViewTransformer {
-    enum Status {
-        case success(_ data: ProjectFactsViewData.Data)
-        case failure(_ failure: String)
-    }
-}
+extension ProjectFactsViewData.Data: StateType {}
```

Comment:
This is where I'm happy to have this intermediate layer between the `Backend` & the `Frontend`, otherwise this would totally be a red flag as `Backend` types would be leaking into the `Frontend`.